core/src/main/java/com/novoda/accessibility/ActionMenuItem.java

```diff
@@ -0,0 +1,273 @@
+        return false;
+    }
+
+    @Override
+    public ActionMenuItem setChecked(boolean checked) {
+        if (checked) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    @Override
+    public boolean isChecked() {
+        return false;
+    }
+
+    @Override
+    public ActionMenuItem setVisible(boolean visible) {
+        if (!visible) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    /**
+     * For our case, a menu item will always/only be visible if enabled.
+     */
+    @Override
+    public boolean isVisible() {
+        return isEnabled();
+    }
+
+    @Override
+    public ActionMenuItem setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        return this;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @Override
+    public boolean hasSubMenu() {
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public SubMenu getSubMenu() {
+        return null;
+    }
+
+    @Override
+    public ActionMenuItem setOnMenuItemClickListener(OnMenuItemClickListener menuItemClickListener) {
+        this.menuItemClickListener = menuItemClickListener;
+        return this;
+    }
+
+    @Nullable
+    @Override
+    public ContextMenu.ContextMenuInfo getMenuInfo() {
+        return null;
+    }
+
+    @Override
+    public void setShowAsAction(int actionEnum) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setShowAsActionFlags(int actionEnum) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setActionView(View view) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setActionView(int resId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public View getActionView() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setActionProvider(ActionProvider actionProvider) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Nullable
+    @Override
+    public ActionProvider getActionProvider() {
+        return null;
+    }
+
+    @Override
+    public boolean expandActionView() {
+        return false;
+    }
+
+    @Override
+    public boolean collapseActionView() {
+        return false;
+    }
+
+    @Override
+    public boolean isActionViewExpanded() {
+        return false;
+    }
+
+    @Override
+    public ActionMenuItem setOnActionExpandListener(OnActionExpandListener listener) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns true if action was invoked, else false.
+     */
+    boolean invokeAction() {
+        if (isEnabled() && menuItemClickListener != null) {
+            menuItemClickListener.onMenuItemClick(this);
```

Comment:
why not `return menuItemClickListener.onMenuItemClick(this);`?

Reply:
commit c231b6a3156aec9897c80f5c2fbe2207bddd65ee
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Apr 3 13:12:04 2018 +0100

    Return result of menuitemclicklistener